pkg/sensors/sensors.go

```diff
@@ -2,21 +2,31 @@ package sensors
 
 import (
 	"encoding/json"
+	"errors"
 	"fmt"
 	"net/http"
+	"syscall"
 )
 
 func CollectSensorData(ip, port string) (map[string]any, error) {
 	resp, err := http.Get(fmt.Sprintf("http://%s:%s", ip, port))
+	connectionRefused := errors.Is(err, syscall.ECONNREFUSED)
+	if err != nil && connectionRefused {
+		err := errors.New("Error requesting sensor data. The API doesn't seem to be working properly. Check its availability")
```

Comment:
it`s not necessarily a change request, but a hint for future projects. Error messages in go should not start with capital letters. In our case, based on how our logs works it is not a problem 

https://github.com/golang/go/wiki/CodeReviewComments#error-strings

Reply:
Thank you for the advice